R/pkgs/inference/R/functions.R

```diff
@@ -544,3 +546,195 @@ iterateAccept <- function(ll_ref, ll_new) {
   }
   return(FALSE)
 }
+
```

Comment:
Not for this PR, but do you mind splitting these off into a different file?

Reply:
do you mean the stuff under the heading "Extra functions for MCMC diagnosis"? if so im happy to do that. they are almost-parallels of functions that are already in here, which is why I put them here, but no problem to move